external-crates/move/crates/move-compiler/src/sui_mode/linters/custom_rules/constant_naming.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//! `ConstantNamingVisitor` enforces a naming convention for constants in Move programs,
+//! requiring them to follow an ALL_CAPS_SNAKE_CASE format. This lint checks each constant's name
+//! within a module against this convention.
+use crate::{
+    diag,
+    diagnostics::codes::{custom, DiagnosticInfo, Severity},
+    shared::{program_info::TypingProgramInfo, CompilationEnv},
+    sui_mode::linters::{LinterDiagCategory, LINTER_DEFAULT_DIAG_CODE, LINT_WARNING_PREFIX},
+    typing::{ast as T, visitor::TypingVisitor},
+};
+use move_ir_types::location::Loc;
+use move_symbol_pool::Symbol;
+
+/// Diagnostic information for constant naming violations.
+const CONSTANT_NAMING_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::ConstantNaming as u8,
+    LINTER_DEFAULT_DIAG_CODE,
+    "Constant name should be in all caps and snake case.",
+);
+
+pub struct ConstantNamingVisitor;
+
+impl TypingVisitor for ConstantNamingVisitor {
+    fn visit(
+        &mut self,
+        env: &mut CompilationEnv,
+        _program_info: &TypingProgramInfo,
+        program: &mut T::Program_,
+    ) {
+        for module_def in program
+            .modules
+            .iter()
+            .filter(|(_, _, mdef)| !mdef.attributes.is_test_or_test_only())
+        {
```

Comment:
I think its fine to lint this in test modules

Reply:
Checkout new PR for this lint #16819